src/pkg/utils/wait.go

```diff
@@ -93,16 +95,25 @@ func ExecuteWait(waitTimeout, waitNamespace, condition, kind, identifier string,
 			spinner.Updatef(existMsg)
 			// Check if the resource exists.
 			zarfKubectlGet := fmt.Sprintf("%s tools kubectl get %s %s %s", zarfCommand, namespaceFlag, kind, identifier)
-			if stdout, stderr, err := exec.Cmd(shell, append(shellArgs, zarfKubectlGet)...); err != nil {
+			stdout, stderr, err := exec.Cmd(shell, append(shellArgs, zarfKubectlGet)...)
+			if err != nil {
 				message.Debug(stdout, stderr, err)
 				continue
 			}
 
+			// If there was no identifier provided and no resources of the given kind were found,
+			// return an error. We don't want to return an error if an identifier
+			// was provided because the resource may be in the process of being created.
+			resourceNotFound := strings.Contains(stderr, "No resources found") && identifier == ""
+			if resourceNotFound {
+				return fmt.Errorf("unable to find resource of kind %q in the cluster", kind)
```

Comment:
this logic should be the same as 100-101 since it could still be in the process of being created and we should wait for it (if the wait times out then we should fail)

Reply:
updated